lib/data/kinetic/backend.js

```diff
@@ -0,0 +1,159 @@
+
+let sequence = 1;
+
+const kDrives = {
+    port: config.kinetic ? config.kinetic.port : 8123,
+    host: config.kinetic ? config.kinetic.host : 'localhost',
+};
+
+function propError(pdu) {
+    const statusCode = pdu.getStatusCode();
+    if (statusCode !== kinetic.errors.SUCCESS) {
+        if (statusCode === kinetic.errors.NOT_FOUND) {
+            return errors.ObjNotFound;
+        }
+        return pdu.getErrorMessage();
+    }
+    return undefined;
+}
+
+function newSocket(callback) {
+    const socket = new net.Socket({ allowHalfOpen: false }).pause();
+    socket.setKeepAlive(true);
+    socket.unref();
+    return socket.connect(kDrives, () => {
+        kinetic.streamToPDU(socket, (err, pdu) => {
+            if (err) {
+                return callback(err);
+            }
+            const err1 = propError(pdu);
+            return callback(err1, socket);
+        });
+    });
+}
+
+function putKinetic(socket, value, options, callback) {
+    const key = crypto.randomBytes(20);
+    const obj = Buffer.concat(value);
+    const tag = crypto.createHmac('sha1', 'asdfasdf').update(obj).digest();
+    const pdu = new kinetic.PutPDU(sequence, key, obj.length, tag, options);
+    ++sequence;
+
+    const header = pdu.read();
+    const len = header.length + obj.length;
+
+    socket.write(Buffer.concat([header, obj], len), err => {
+        if (err) {
+            return callback(err);
+        }
+        kinetic.streamToPDU(socket, (err, pdu) => {
+            const err1 = propError(pdu);
+            socket.destroy();
+            return callback(err || err1, key);
+        });
+        return undefined;
+    });
+}
+
+function getKinetic(socket, pdu, callback) {
+    let count = 0;
+    const chunkSize = pdu.getChunkSize();
+    const value = [];
+    const err = propError(pdu);
+    socket.resume();
+    socket.on('data', chunk => {
+        value.push(chunk);
+        count += chunk.length;
+        if (count === chunkSize) {
+            socket.end();
+        }
+    }).on('end', () => {
+        socket.destroy();
+        return callback(err, new stream.Readable({
+            read() {
+                this.push(Buffer.concat(value));
+                this.push(null);
+            },
+        }));
+    });
+}
+
+function deleteKinetic(socket, key, callback) {
+    const pdu = new kinetic.DeletePDU(sequence, key);
+    ++sequence;
+    socket.write(pdu.read(), err => {
+        if (err) {
+            return callback(err);
+        }
+        kinetic.streamToPDU(socket, (err, pdu) => {
+            if (err) {
+                return callback(err);
+            }
+            const err1 = propError(pdu);
+            socket.destroy();
+            return callback(err || err1);
+        });
+        return undefined;
+    });
+}
+
+const backend = {
+    put: function putK(request, size, keyContext, reqUids, callback) {
+        const value = [];
+        request.on('data', data => {
+            value.push(data);
```

Comment:
Yes that is what we wrote in the commit message that for now we are not handling more than 1mB files we are currently on it.
Our idea was to split into multiple 1MB files.